lib/Doctrine/ORM/Event/LoadClassMetadataEventArgs.php

```diff
@@ -30,12 +32,40 @@
 class LoadClassMetadataEventArgs extends BaseLoadClassMetadataEventArgs
 {
     /**
+     * @param ClassMetadata $classMetadata
+     * @param EntityManager $entityManager
+     */
+    function __construct(ClassMetadata $classMetadata, EntityManager $entityManager)
+    {
+        /*
+        We use our own constructor here to enforce type-hinting requirements,
+        since both inputs are specialized subclasses compared to what the super-
+        class is willing to accept.
+
+        In particular, we want to have EntityManager rather than ObjectManager.
+        */
```

Comment:
The comment answers the question "Why Even Have This?" that people will ask... but only if they're looking at the seemingly-pointless guts of the method.

From outside the class, nobody needs to know *why* the constructor exists, they just need to know how to use it.